### PR TITLE
Fix issue with external GW setup + run CI with `--disable-snat-multiple-gws` on and off

### PR DESCRIPTION
**Pull request by alexanderConstantinescu**
**- What this PR does and why is it needed**

Two issues:

1. This fixes an issue with the external GW setup when the flag  `--disable-snat-multiple-gws` is provided, namely: the conditional checks `(len(nsInfo.routingExternalGWs.gws) != 0 || len(nsInfo.routingExternalPodGWs) != 0)` was causing this bloc of code to be practically be considered dead code: https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/namespace.go#L249-L261 <br />
`nsInfo.routingExternalGWs` can only be set in `addExternalGWsForNamespace`, which comes after that conditional check. Moreover, `nsInfo.routingExternalPodGWs` concerns itself with unique pod setup for multiple gateway traffic and has dedicated pod labels for that configuration, those have no relation to the namespace labels (which might be independently [in]correct). Given that a namespace includes that pod, this should override it. All of this results in the fact that if the annotation is added to the namespace: this condition will block the deletion of the "per-pod snat" setup, which is required to make multiple gateway traffic work. 

2. OpenShift will run clusters with `--disable-snat-multiple-gws` provided, since it's a rather important change w.r.t how ovnkube runs in shared GW mode, enable it in CI. 

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by trozet** [comment]
looks similar to what @fedepaol is trying to fix in https://github.com/ovn-org/ovn-kubernetes/pull/2272 right?

File: go-controller/pkg/ovn/namespace.go [hunk 1]
```diff
@@ -245,7 +245,7 @@ func (oc *Controller) updateNamespace(old, newer *kapi.Namespace) {
 	if gwAnnotation != oldGWAnnotation || newBFDEnabled != oldBFDEnabled {
 		// if old gw annotation was empty, new one must not be empty, so we should remove any per pod SNAT
 		if oldGWAnnotation == "" {
-			if config.Gateway.DisableSNATMultipleGWs && (len(nsInfo.routingExternalGWs.gws) != 0 || len(nsInfo.routingExternalPodGWs) != 0) {
```
Comment (trozet) [inline go-controller/pkg/ovn/namespace.go:248]: change seems logical, but you have to consider the external pod gw path I think. If a pod is added to serve as an external gateway, we also need to remove the SNAT then.

**Review by trozet — CHANGES_REQUESTED** [review]

File: go-controller/pkg/ovn/namespace.go (hunk 1 above)
Comment (alexanderConstantinescu) [reply go-controller/pkg/ovn/namespace.go:248]: Yes, that is what #2272 fixes.

**Comment by alexanderConstantinescu** [comment]
> looks similar to what @fedepaol is trying to fix in #2272 right?

It's similar, but different. #2272 is fixing the problem "the pod is annotated to be used as an external GW route, but we are not removing its dedicated SNAT on the GR". This is fixing the problem: "the namespace is annotated to be used an external GW route, but we are not removing the SNATs on the GR". 

Mind you, all of this has been completely untested until now, since we were not running CI with `--disable-snat-multiple-gws`...which is what this PR now does. As you can see from the CI result on this PR: https://github.com/ovn-org/ovn-kubernetes/pull/2280/checks?check_run_id=2914505893 the tests:

```
[Fail] e2e non-vxlan external gateway through a gateway pod Should validate TCP/UDP connectivity to an external gateway's loopback address via a pod with external gateway annotations enabled [It] UDP ipv4 
/home/runner/work/ovn-kubernetes/ovn-kubernetes/test/e2e/external_gateways.go:885

[Fail] e2e non-vxlan external gateway through a gateway pod Should validate TCP/UDP connectivity to an external gateway's loopback address via a pod with external gateway annotations enabled [It] UDP ipv4 
/home/runner/work/ovn-kubernetes/ovn-kubernetes/test/e2e/external_gateways.go:885
```

Still fail. And that is what #2272 will fix.

**Comment by fedepaol** [comment]
@alexanderConstantinescu I think we can close this, as we discussed I stole the commits and added them to https://github.com/ovn-org/ovn-kubernetes/pull/2272

**Comment by alexanderConstantinescu** [comment]
> @alexanderConstantinescu I think we can close this, as we discussed I stole the commits and added them to #2272

OK, thanks a lot!

**Pull request closed without merging**